### ISSUE TITLE
Additional outputs missing in PoseNet

### ISSUE THREAD

**Issue by DanielCoelho112** [DanielCoelho112/localization_end_to_end#60]
Hi @miguelriemoliveira and @pmdjdias,

When training PoseNet I was experiencing a lot of overfitting that not even high dropout rates and high L2 regularizations were able to reduce.  Today I went to check the PoseNet paper again and found that they were still using 2 additional outputs to stabilize the google net.

The github implementations of PoseNet that I found didn't consider this. I think we can achieve even better results with this feature. I'll implement it and let you know if it helped.

**Comment by miguelriemoliveira** [comment]
Hi @DanielCoelho112 ,

I hope that is the cause for te over fitting. Let me know how it goes.

**Comment by DanielCoelho112** [comment]
Done. It helped solve the overfitting.

**Comment by miguelriemoliveira** [comment]
An image of the training please !!!!

**Comment by DanielCoelho112** [comment]
![losses](https://user-images.githubusercontent.com/73063947/172587120-5dff2b04-8278-43dc-bb06-24e4612d78c2.png)

Although it looks like there is still some overfitting the reason is the following:

Loss_train = Loss_final + 0.3(Loss_auxiliar1 + Loss_auxiliar2)

Loss_test = Loss_final

So, in training, we are adding 3 negative terms, and in test, we are only adding one.  I say that we are free of overfitting because our test loss keeps decreasing and following test loss.

**Comment by miguelriemoliveira** [comment]
Looks good. Actually, it looked like you could've continued the training a bit more.

**Comment by pmdjdias** [comment]
Agree… “Uma imagem vale mil palavras” 😉!

--Paulo
